File: GeeksforGeeks/Basic/Sum of numbers in string .cpp
// { Driver Code Starts
// C++ program to calculate sum of all numbers present
// in a string containing alphanumeric characters
#include <iostream>
using namespace std;


 // } Driver Code Ends





// Function to calculate sum of all numbers present
// in a string containing alphanumeric characters
int findSum(string str)
{
	
	// Your code here
	 int n = str.length(),sum = 0,curNum = 0;

    for(int i = 0 ; i < n ; i++) {
        char cur = str[i];

        if(cur >= 48 and cur <= 57)
            curNum = (curNum * 10) + (cur - '0');
        else {
            sum += curNum;
            curNum = 0;
        }

    }
    
    if(curNum != 0)
        sum += curNum;

    return sum;
	
}


// { Driver Code Starts.

// Driver code
int main()
{
	// input alphanumeric string
	int t;
	cin>>t;
	while(t--)
	{
	    string str;
	    cin>>str;
	    cout << findSum(str);
        cout<<endl;
	}
	return 0;
}
  // } Driver Code Ends

File: GeeksforGeeks/Basic/Java/Sum of numbers in string.java
// { Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
	public static void main (String[] args) throws IOException{
	    BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
	    StringTokenizer st=new StringTokenizer(br.readLine());
	    int t=Integer.parseInt(st.nextToken());
	    while(t-->0){
	        String s=br.readLine();
	        
	        SumNumbers obj = new SumNumbers();
	        
	        System.out.println(obj.findSum(s));
	    }
		
	}
}
// } Driver Code Ends


class SumNumbers{
    // Function to find the sum of all
    // the numbers in the given string s
    public static long findSum(String s){
        
        // your code here
        long sum = 0;
        String str = "";
        s = s + " ";
        
        for(int i = 0; i < s.length(); i++)
        {
            char ch = s.charAt(i);
            if(Character.isDigit(ch)) 
            str += ch;
            else if(str.length() > 0)
            {sum += Integer.parseInt(str);
             str = "";}
        }
        return sum;
        
        
    }
    
}